src/Libraries/Microsoft.AspNetCore.HeaderParsing/README.md

```diff
@@ -1 +1,130 @@
-README
+# Microsoft.AspNetCore.HeaderParsing
+
+This package provides services to for strongly typed header parsing and value caching.
+
+In particular:
+
+- There are many different headers with subtle format differences that are hard to parse correctly in all cases. This package provides predefined parsers for commonly used headers.
+- Your application may parse the same header value multiple times per request. This package automatically caches the parsed values of these headers to make your application faster.
+- This package also provides logging and metering of bad header values.
+
+## Getting started
+
+### Install the package
+
+From the command-line:
+
+```dotnetcli
+dotnet add package Microsoft.AspNetCore.HeaderParsing
+```
+
+Or directly in the C# project file:
+
+```xml
+<ItemGroup>
+  <PackageReference Include="Microsoft.AspNetCore.HeaderParsing" Version="*" />
+</ItemGroup>
+```
+## Usage Example
+
+### Registering the services
+
+The services can be registered using one of the `AddHeaderParsing` overloads:
+
+```csharp
+public static IServiceCollection AddHeaderParsing(this IServiceCollection services);
+public static IServiceCollection AddHeaderParsing(this IServiceCollection services, Action<HeaderParsingOptions> configuration);
```

Comment:
Shouldn't this be 
```suggestion
public static IServiceCollection AddHeaderParsing(this IServiceCollection services, Action<HeaderParsingOptions> configure);
```

Reply:
Thanks, I will report it to the API review. And the other instances I found BTW.

Reply:
I am merging this suggestion, but the API is wrong, this will need to be fixed.

Reply:
Want to just fix the API in this PR?

Reply:
Done